xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -4158,6 +4164,27 @@ public void readDocMetaFromForm(EditForm eform, XWikiContext context) throws XWi
         if (eform.getHidden() != null) {
             setHidden("1".equals(eform.getHidden()));
         }
+
+        if (eform.getActivateRequiredRights() != null) {
+            if (Objects.equals(eform.getActivateRequiredRights(), "1")) {
+                this.requiredRightsActivated = true;
+                setMetaDataDirty(true);
```

Comment:
The metadata dirty flag should be changed only if the required right flag has changed no?